lib/collection/src/common/snapshots_manager.rs

```diff
@@ -9,19 +11,51 @@ use crate::common::sha_256::hash_file;
 use crate::operations::snapshot_ops::{
     get_checksum_path, get_snapshot_description, SnapshotDescription,
 };
-use crate::operations::types::CollectionResult;
+use crate::operations::snapshot_s3_ops;
+use crate::operations::types::{CollectionError, CollectionResult};
 
-#[derive(Clone, Deserialize, Debug)]
+#[derive(Clone, Deserialize, Debug, Default)]
+pub struct SnapShotsConfig {
+    pub snapshots_storage: SnapshotsStorageConfig,
+    pub s3_config: Option<S3Config>,
+}
+
+#[derive(Clone, Debug, Default)]
+pub enum SnapshotsStorageConfig {
+    #[default]
+    Local,
+    S3,
+}
+
+impl<'de> Deserialize<'de> for SnapshotsStorageConfig {
+    fn deserialize<D>(deserializer: D) -> Result<SnapshotsStorageConfig, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let s: String = Deserialize::deserialize(deserializer)?;
+        match s.as_str() {
+            "local" => Ok(SnapshotsStorageConfig::Local),
+            "s3" => Ok(SnapshotsStorageConfig::S3),
+            _ => Err(serde::de::Error::custom(
+                "Invalid snapshots_storage. Use 'local' or 's3'",
+            )),
+        }
+    }
```

Comment:
Why custom deserializer is required in this case?

If it is required, how to ensure that we won't forget to add a support for a new type here if enum is changed?